pinot-query-runtime/src/test/resources/queries/MathFuncs.json

```diff
@@ -0,0 +1,439 @@
+        "comment": "should round on the integer but we return a floating point",
+        "description": "test divide on integer columns",
+        "sql": "SELECT intCol / longCol FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "should round on the integer but we return a floating point",
+        "description": "test divide on int literal with columns",
+        "sql": "SELECT intCol / 10 FROM {numTbl}"
+      },
+      {
+        "description": "test divide on double literal with columns",
+        "sql": "SELECT intCol / 1.2 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "should round on the integer but we return a floating point",
+        "description": "test divide on literals",
+        "sql": "SELECT 3 / 5 FROM {numTbl}"
+      },
+      {
+        "description": "test divide on literals",
+        "sql": "SELECT 3 / 2.4 FROM {numTbl}"
+      },
+      {
+        "description": "test divide on floating point",
+        "sql": "SELECT doubleCol / floatCol FROM {numTbl}"
+      },
+      {
+        "description": "test divide on floating point with literal",
+        "sql": "SELECT doubleCol / 2 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "should throw infinity but we throw an exception",
+        "description": "test divide by zero",
+        "expectedException": ".*Division by zero.*",
+        "sql": "SELECT doubleCol / 0 FROM {numTbl}"
+      },
+      {
+        "description": "test divide on doubleCol with intCol",
+        "sql": "SELECT doubleCol * intCol FROM {numTbl}"
+      }
+    ]
+  },
+  "mod": {
+    "tables": {
+      "numTbl": {
+        "schema": [
+          {"name": "intCol", "type": "INT"},
+          {"name": "longCol", "type": "LONG"},
+          {"name": "doubleCol", "type": "DOUBLE"},
+          {"name": "floatCol", "type": "FLOAT"}
+        ],
+        "inputs": [
+          [0, 3, 0.1, 3.2],
+          [123, 321, 4.2, 3.0],
+          [-456, -2, 1.1, 7.7],
+          [123, -456, 3.6, 9.1]
+        ]
+      }
+    },
+    "queries": [
+      {
+        "ignored": true,
+        "comment": "we are returning -0.0 for some reason",
+        "description": "test mod on integer columns",
+        "sql": "SELECT intCol % longCol FROM {numTbl}"
+      },
+      {
+        "description": "test mod on int literal with columns",
+        "sql": "SELECT intCol %  10 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "should return 0 for 0 value but we are not",
+        "description": "test mod on floating point literal with columns",
+        "sql": "SELECT intCol %  1.2 FROM {numTbl}"
+      },
+      {
+        "comment": "should round on the integer but we return a floating point",
+        "description": "test mod on literals",
+        "sql": "SELECT 3 %  5 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "diff due to floating point comparison",
```

Comment:
can we fix the test framework to compare floating points properly? we can compare within .0001 error bound is probably fine

Reply:
+1

Reply:
i think there's already a fix so let's see if these passes now

Reply:
Done